scripts/meteor-recalibrate-movement.py

```diff
@@ -0,0 +1,132 @@
+to the calibration metadata format and update the stage-bare component with the new values.
+
+run as:
+python3 ./scripts/meteor-recalibrate-movement.py 
+
+and follow the instructions in the terminal.
+
+'''
+
+# TODO: update the yaml file with these values too
+# TODO: create a plugin for this
+
+import math
+from odemis import model
+from odemis.acq.move import has_axis_move
+from odemis.util.driver import isInRange
+from Pyro4.errors import CommunicationError
+
+def main():
+
+    print(f"METEOR Stage Movement Calibration Updater")
+    print(f"-" * 20)
+    
+    # get stage-bare component
+    try:
+        stage = model.getComponent(role="stage-bare")
+    except CommunicationError as e:
+        print("ERROR: Could not connect to odemis. Is odemis running?")
+        return
+
+    print(f"Current stage-bare component: {stage}")    
+    print(f"-" * 20)
+    
+    # check which movement mode the calibration data is intended for, if not FIB <-> METEOR, warn user.
+    stage_md = stage.getMetadata()
+    fm_pos_active = stage_md[model.MD_FAV_FM_POS_ACTIVE]
+    sem_pos_active = stage_md[model.MD_FAV_SEM_POS_ACTIVE]
+    has_rz = has_axis_move(sem_pos_active, fm_pos_active, "rz") # returns False if FIB <-> METEOR movement
+
+    if has_rz:
+        sem_rz = f"{math.degrees(sem_pos_active['rz']):02f}"
+        flm_rz = f"{math.degrees(fm_pos_active['rz']):02f}"
+        print(f"""ERROR: The current calibration data is for SEM <-> METEOR movement.
+              \nThe current rotation positions are: FIBSEM: {sem_rz} deg, METEOR: {flm_rz} deg
+              \nThis script is intended for FIB <-> METEOR movement (movement with no rotation). 
+              \nIf you wish to continue, please update the calibration data in the yaml file to FIB <-> METEOR movement. 
+              \nYou will need to restart odemis for the changes to take effect.""")
+        return
+
+    print(f"Valid calibration data found for FIB <-> METEOR movement. Continuing...")
+    print('-' * 20)
+
+    print(f"Calibration values are the distance between the FIB and METEOR positions in the x and y directions.")
+    print(f"Current calibration values: {stage_md[model.MD_POS_COR]}")
+
+    # get imaging ranges
+    sem_active_range = stage_md[model.MD_SEM_IMAGING_RANGE]
+    flm_active_range = stage_md[model.MD_FM_IMAGING_RANGE]
+    working_range = {"x": [0, 0], "y": [0, 0]}
+
+    for axis in ["x", "y"]: 
+        margin = (sem_active_range[axis][1] - sem_active_range[axis][0]) * 0.05 # 5% of SEM imaging range
+        x0 = flm_active_range[axis][0] - sem_active_range[axis][0] - margin     # minimum move sem -> flm
+        x1 = flm_active_range[axis][1] - sem_active_range[axis][0] + margin     # maximum move sem -> flm
+        working_range[axis] = [x0, x1]
+ 
+    print(f"""Calibration data should be within the ranges: 
+          \nx range: {[f'{v:4f}' for v in working_range['x']]}
+          \ny range: {[f'{v:4f}' for v in working_range['y']]} 
```

Comment:
Please don't mix `\n` and triple quote strings. The whole point of triple quote strings is that you don't need `\n`, and you can write down directly as it'll appear. So just explicity add an empty line, or drop the `\n`.